core/server/src/test/java/alluxio/worker/netty/DataServerBlockReadHandlerTest.java

```diff
@@ -0,0 +1,304 @@
+  private RPCProtoMessage buildReadRequest(long offset, long len) {
+    Protocol.ReadRequest readRequest =
+        Protocol.ReadRequest.newBuilder().setId(mBlockId).setOffset(offset).setLength(len)
+            .setLockId(1L).setType(Protocol.RequestType.ALLUXIO_BLOCK).build();
+    return new RPCProtoMessage(readRequest, null);
+  }
+
+  /**
+   * Checks all the read responses.
+   */
+  private void checkAllReadResponses() {
+    int timeRemaining = Constants.MINUTE_MS;
+    boolean eof = false;
+    long checksumActual = 0;
+    while (!eof && timeRemaining > 0) {
+      Object readResponse = null;
+      while (readResponse == null && timeRemaining > 0) {
+        readResponse = mChannel.readOutbound();
+        CommonUtils.sleepMs(10);
+        timeRemaining -= 10;
+      }
+      DataBuffer buffer = checkReadResponse(readResponse, Protocol.Status.Code.OK);
+      eof = buffer == null;
+      if (buffer != null) {
+        if (buffer instanceof DataNettyBufferV2) {
+          ByteBuf buf = (ByteBuf) buffer.getNettyOutput();
+          while (buf.readableBytes() > 0) {
+            checksumActual += BufferUtils.byteToInt(buf.readByte());
+          }
+          buf.release();
+        } else {
+          Assert.assertTrue(buffer instanceof DataFileChannel);
+          ByteBuffer buf = buffer.getReadOnlyByteBuffer();
+          byte[] array = new byte[buf.remaining()];
+          buf.get(array);
+          for (int i = 0; i < array.length; i++) {
+            checksumActual += BufferUtils.byteToInt(array[i]);
+          }
+        }
+      }
+    }
+    Assert.assertEquals(mChecksum, checksumActual);
+    Assert.assertTrue(eof);
+  }
+
+  /**
+   * Checks the read response message given the expected error code.
+   *
+   * @param readResponse the read response
+   * @param codeExpected the expected error code
+   * @return the data buffer extracted from the read response
+   */
+  private DataBuffer checkReadResponse(Object readResponse, Protocol.Status.Code codeExpected) {
+    Assert.assertTrue(readResponse instanceof RPCProtoMessage);
+
+    Object response = ((RPCProtoMessage) readResponse).getMessage();
+    Assert.assertTrue(response instanceof Protocol.Response);
+    Assert.assertEquals(codeExpected, ((Protocol.Response) response).getStatus().getCode());
+    return ((RPCProtoMessage) readResponse).getPayloadDataBuffer();
+  }
+
+  /**
+   * Waits for one read response messsage.
+   *
+   * @return the read response
+   */
+  private Object waitForOneResponse() {
+    Object writeResponse = null;
+    int timeRemaining = Constants.MINUTE_MS;
+    while (writeResponse == null && timeRemaining > 0) {
+      writeResponse = mChannel.readOutbound();
+      CommonUtils.sleepMs(10);
+      timeRemaining -= 10;
+    }
+    return writeResponse;
+  }
+
+  /**
+   * Waits for the channel to close.
+   */
+  private void waitForChannelClose() {
+    int timeRemaining = Constants.MINUTE_MS;
+    while (timeRemaining > 0 && mChannel.isOpen()) {
```

Comment:
there is no sleep in the body, and `timeRemaining` is not updated? Could this use `CommonUtils.waitFor`?